snapcraft/internal/elf.py

```diff
@@ -107,77 +118,49 @@ def __init__(self, *, path: str) -> None:
         """
         self.path = path
         self.dependencies = set()  # type: Set[Library]
-        self.interp, self.symbols = self._extract_readelf(path)
+        self.interp, self.soname, self.needed = self._extract(path)
 
-    def _extract_readelf(self, path) -> Tuple[str, List[Symbol]]:
+    def _extract(self, path) -> Tuple[str, str, Dict[str, NeededLibrary]]:  # noqa: C901,E501
```

Comment:
Well this seems a bit like cheating. Especially C901.

Reply:
I initially tried wrapping the line, and pyflakes ignored the annotation: it seems that it must appear on the same line as the beginning of the method definition.

As for the method itself, it was simpler before adding the str/bytes checks to support xenial's old pyelftools.  I'll see what I can do.